src/public/script.js

```diff
@@ -4,140 +4,131 @@ const limitSelect = document.querySelector('#limit-select');
 const pingUrl = '/api/ping';
-const setLimitUrl = '/api/set-limit';
-let responses = {
-    successful: 0,
-    limited: 0,
-    unknown: 0
-};
-let timeout;
-
-const onLimitSelectChange = async e => {
-    try {
-        await axios.put(setLimitUrl, { limit: e.target.value });
-    } catch (err) {
-        console.error(err);
-    }
-};
 
-const onSendButtonClick = async e => {
+const onSendButtonClick = e => {
     e.preventDefault();
 
-    if (!timeout) {
-        let interval;
-        let counter = 10;
-
-        limitSelect.disabled = true;
-
-        timeout = setTimeout(() => {
-            resetButton.disabled = false;
-            sendButton.disabled = true;
-            limitSelect.disabled = false;
-            clearInterval(interval);
-            timerDiv.innerHTML = 'Done, reset your stats';
-        }, 10 * 1000);
-
-        timerDiv.innerHTML = `${counter} seconds left`;
-
-        interval = setInterval(() => {
-            counter--;
-            timerDiv.innerHTML = `${counter} seconds left`;
-        }, 1000);
+    const requestLimit = parseInt(limitSelect.value);
+    let singleRequests = 0;
+    let doubleRequests = 0;
+
+    switch (requestLimit) {
+        case 5:
+            singleRequests = 5;
+            break;
+        case 10:
+            singleRequests = 10;
+            break;
+        case 15:
+            singleRequests = 5;
+            doubleRequests = 5;
+            break;
+        case 20:
+            doubleRequests = 10;
+            break;
+        default:
+            break;
     }
 
-    let results = [];
-    let calls = [];
+    let successfullRequests = 0;
+    let blockedRequests = 0;
+    let counter = 10;
+    let interval;
+
+    sendButton.disabled = true;
+    limitSelect.disabled = true;
 
     const callPing = async () => {
         try {
-            const response = await axios.get(pingUrl, {
+            await axios.get(pingUrl, {
                 headers: {
                     'Cache-Control': 'no-cache, no-store, must-revalidate',
                     Pragma: 'no-cache',
                     Expires: '0'
                 }
             });
 
-            results.push(response);
+            successfullRequests++;
         } catch (err) {
-            results.push(err.response);
+            blockedRequests++;
         }
     };
 
-    for (let i = 0; i < 2; i++) {
-        calls.push(callPing());
-    }
+    timerDiv.innerHTML = `${counter} seconds left`;
 
-    await Promise.all(calls);
+    interval = setInterval(async () => {
+        let requests = [];
+        counter--;
 
-    results.forEach(result => {
-        if (!result || !result.status || (result.status !== 200 && result.status !== 429)) {
-            responses.unknown++;
+        if (counter) {
+            timerDiv.innerHTML = `${counter} seconds left`;
         }
-        if (result.status === 200) {
-            responses.successful++;
+
+        if (doubleRequests) {
+            for (let i = 0; i < 2; i++) {
```

Comment:
cannot you just add here some variable?

for (let i = 0; i < variable; i++)

and remove all these doubleRequests, singleRequests variables?

Reply:
eventually - just use setInterval for 500 ms; 

const requestsToSend = 20; 
const whenToSendTick = 20 / requestsToSend; // for 5 requests, send each 4 ticks, for 10 send each 2 ticks, for 20 send each 1 tick
const tick = 0; 

if (tick % whenToSendTick === 0) {
   // send request
}

Reply:
done